code/scripts/espurna_utils/bsec.py

```diff
@@ -0,0 +1,44 @@
+import os
+import pathlib
+
+def bsec_inject_patcher(env):
+    bsec_builder = next((builder for builder in env.GetLibBuilders() if builder.name == 'BSEC Software Library'), None)
+
+    if bsec_builder is None:
+        return
+
+    bsec_dir = os.path.join(bsec_builder.src_dir, 'esp8266')
+    patchflag_path = os.path.join(bsec_dir, '.patched')
+
+    env.Append(LIBPATH=[bsec_dir])
+    env.Append(LIBS=["algobsec"])
+
+    if os.path.isfile(patchflag_path):
+      print("Archive libalgosec.a has already been patched")
+      return
+
+    script = env.Command(
+        os.path.join(bsec_dir, "libalgobsec.a.out"),
+        os.path.join(bsec_dir, "libalgobsec.a"),
+        [
+            'mv -vn libalgobsec.a libalgobsec.a.orig',
```

Comment:
re 1) Windows
- cmd.exe only has `rename` or `move`, not `mv`
- `touch` is perhaps `echo > .patched`

$AR is there, no issue calling that

Or, just use python as common way to handle that instead of shell call

Reply:
Already have everything running under native python except `$AR` which would be part of the xtensa toolchain available on Windows, as you've mentioned.